corehq/apps/linked_domain/models.py

```diff
@@ -83,11 +83,13 @@ def update_last_pull(self, model, user_id, date=None, model_detail=None):
     def save(self, *args, **kwargs):
         super(DomainLink, self).save(*args, **kwargs)
         from corehq.apps.linked_domain.dbaccessors import (
-            get_upstream_domain_link,
+            get_domain_link,
             get_linked_domains,
+            get_upstream_domain_link,
             is_active_downstream_domain,
             is_active_upstream_domain,
         )
+        get_domain_link.clear(self.master_domain, self.linked_domain)
```

Comment:
This is theoretically error-prone. Consider a domain link between domains A and B. For whatever reason, someone pulls up this domain link and changes the link to be between A and C. This method will clear the cache for A and C, but the cache will still believe there is a valid link between A and B.

Reply:
This has been resolved in that I've removed the `get_domain_link` method entirely, but this brings up an interesting thought. As far as available workflows to web users, a domain link cannot be edited in that way. One would need to open a db shell and edit the object directly to achieve this scenario. If something like this comes up in the future, should we build into code the ability to handle edge cases like this?

Reply:
I don't think existing workflows are enough of a safeguard to prevent misuse in the future. We want to design our code to be robust, not knowing how it may be used in the future. There is definitely a valid argument about over-engineering, but if we feel that something like this would be over-engineering, we should at least be leaving comments in the code mentioning that domain links should not be modified in this way, and will break if they are.